### PR TITLE
Support handlers without extracting

### LINKED ISSUE #166: Support handlers without extraction methods
We plan on writing handlers for executable formats such as PE or ELF. It's useful to carve them out of larger chunks but there is no extraction or decompression method required given that the carved out file is all there is to it.

Unblob should support handlers with a `make_extract_command` function that is either undefined or returns an empty list. Whichever solution is best.

### PR DESCRIPTION
**Pull request by kukovecz**
Fixes #166, #147

### CODE REVIEW

File: unblob/models.py [hunk 1]
```diff
@@ -119,9 +119,12 @@ def calculate_chunk(
         """Calculate the Chunk offsets from the Blob and the file type headers."""
 
     @staticmethod
-    @abc.abstractmethod
     def make_extract_command(inpath: str, outdir: str) -> List[str]:
-        """Make the extract command with the external tool, which can be passed for subprocess.run."""
+        """
+        Make the extract command with the external tool, which can be passed for subprocess.run.
+        Return with an empty list of the handler does not supposed to perform extraction.
```
Comment (QKaiser) [inline unblob/models.py:125]: I would use "Returns an empty list if the handler is not supposed to perform extractions."

File: unblob/processing.py [hunk 2]
```diff
@@ -158,13 +159,21 @@ def _process_regular_file(self, task: Task, size: int, result: TaskResult):
...
-                new_dir = extract_with_command(
-                    extract_dir, carved_valid_path, chunk.handler, result
-                )
+                inpath, outdir = get_extract_paths(extract_dir, carved_valid_path)
+                command = chunk.handler.make_extract_command(str(inpath), str(outdir))
+
+                if not command:
+                    logger.debug(
+                        "No need to extract, as this handler does not have extractor",
+                        handler=chunk.handler.NAME,
```
Comment (QKaiser) [inline unblob/processing.py:168]: I would increase the verbosity of this log to 2.

**Review by QKaiser — COMMENTED** [review]
Except for small changes like verbose level and docstrings, this looks good to me. Maybe just  wait for @vlaci or @kissgyorgy to take a look prior to merging tho :)

File: unblob/models.py (hunk 1 above)
Comment (kukovecz) [reply unblob/models.py]: Fixed, thanks!

File: unblob/processing.py (hunk 2 above)
Comment (kukovecz) [reply unblob/processing.py:168]: The verbosity thingy is in a pull request right now, not part of the main branch - but I like the way you are so familiar with it early on! :smile: 

I leave this conversation unresolved and either fix it here if the verbosity stuff merged first, or fix it there if this is merged first.

Comment (QKaiser) [reply unblob/processing.py:168]: Let's do it once verbosity branch gets merged :)

**Review by QKaiser — APPROVED** [review]

**Review by vlaci — APPROVED** [review]

**Pull request merged**